lib/util/src/hse_err.c

```diff
@@ -118,27 +109,42 @@ merr_strerror(merr_t err, char *buf, size_t buf_sz)
 char *
 merr_strinfo(merr_t err, char *buf, size_t buf_sz, size_t *need_sz)
 {
-    ssize_t     sz = 0;
+    ssize_t     sz = 0, ret = 0;
     const char *file = NULL;
 
     if (err) {
         file = merr_file(err);
+
         if (file)
-            sz = snprintf(buf, buf_sz, "%s:%d: ", file, merr_lineno(err));
-        if (sz < 0) {
+            ret = snprintf(buf, buf_sz, "%s:%d: ", file, merr_lineno(err));
+
+        if (ret < 0) {
             sz = strlcpy(buf, "<failed to format error message>", buf_sz);
             goto out;
         }
+
+        sz += ret;
         if (sz >= buf_sz)
             goto out;
 
         sz += merr_strerror(err, buf + sz, buf_sz - sz);
+        if (sz >= buf_sz)
+            goto out;
+
+        ret = snprintf(buf + sz, buf_sz - sz, " (%d)", merr_errno(err));
+        if (ret < 0) {
+            sz = strlcpy(buf, "<failed add errno to error message>", buf_sz);
```

Comment:
"failed to add..."

Reply:
Hmmm, this obliterates the useful error message with something unhelpful just because it couldn't tack on the errno to the end...  I'd get rid of it...

Reply:
This is a case that should almost never happen however.

Reply:
Agreed, but as-is it punishes the caller who passes in a buf that cannot contain the entire message, so rather than get a partial message which is useful, you get back an messages that says the conversion failed but no clue as to why it failed.  I'm not a big fan of the "punishment" approach, give them back as much useful info as we can...

Reply:
Are we guaranteed that the string is stable after snprintf returned an error? their is not indication either way from man. So it is safer to be pessimistic IMO.

Reply:
Good point!  But you could just nul terminate it and you'd be good (i.e., buf[sz] = '\000')